mlibrary/search
# Replace `UNSAFE_componentWillReceiveProps`

# Overview
The `componentWillReceiveProps` lifecycle hook has been deprecated. As a [quick fix](https://github.com/mlibrary/search/commit/b5765af1d7bb9078d21a6315a98a30061655b6b1), React recommended changing the name to `UNSAFE_componentWillReceiveProps`. This pull request officially replaces those lifycycle hooks, and to take it a step further, converts the components to functional components (except URLSearchQueryWrapper), as this is now [standard practice in React](https://react.dev/learn/your-first-component).

## Testing
List instructions on how to test the pull request. Some examples:

- Run the tests to make sure they pass (`npm run test`).
  - Break the new/updated unit tests to make sure they're working properly.
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Go through the site and check to see if anything is broken, or any errors come up.
